state.go

```diff
@@ -405,8 +405,11 @@ func (uc *UpdateCommitState) Handle(ctx *StateContext, c Controller) (State, boo
 	err := c.CommitUpdate()
 	if err != nil {
 		log.Errorf("update commit failed: %s", err)
-		// TODO: should we rollback?
-		return NewUpdateStatusReportState(uc.update, client.StatusFailure), false
+		// we need to perform roll-back here; one scenario is when u-boot fw utils
+		// won't work after update; at this point without rolling-back it won't be
+		// possible to perform new update
+		// as the update was not commited we can safely reboot only
+		return NewRebootState(uc.update), false
```

Comment:
Yes, I can see now. You are *absolutely* right. Here reboot is enough. The only place, where we should use rollback state is when we can not report successful update (which is happening just after commit).